be/src/formats/parquet/scalar_column_reader.cpp

```diff
@@ -342,6 +351,84 @@ StatusOr<bool> RawColumnReader::_page_index_zone_map_filter(const std::vector<co
+                                           TProtocolType::COMPACT, &header));
+    if (length == 0) {
+        bloom_filter_data.resize(header_len + header.numBytes + 1);
+        RETURN_IF_ERROR(
+                _opts.file->read_at_fully(offset + header_len, bloom_filter_data.data() + header_len, header.numBytes));
+    }
+    if (get_chunk_metadata()->meta_data.__isset.statistics &&
+        get_chunk_metadata()->meta_data.statistics.__isset.null_count) {
+        if (get_chunk_metadata()->meta_data.statistics.null_count > 0) {
+            bloom_filter_data.back() = 1;
+        } else {
+            bloom_filter_data.back() = 0;
+        }
+    } else if (get_column_parquet_field()->is_nullable) {
+        bloom_filter_data.back() = 1; //set has null as default, to avoid `column is null` to filter the group.
+    } else {
+        bloom_filter_data.back() = 0;
+    }
+
+    return bloom_filter.init(bloom_filter_data.data() + header_len, header.numBytes + 1, HashStrategyPB::XXHASH64);
+}
+
+StatusOr<bool> RawColumnReader::_row_group_bloom_filter(const std::vector<const ColumnPredicate*>& predicates,
+                                                        CompoundNodeType pred_relation, const TypeDescriptor& col_type,
+                                                        const uint64_t rg_first_row, const uint64_t rg_num_rows) const {
+    bool filtered = false;
+    auto& column_metadata = get_chunk_metadata()->meta_data;
+    bool has_bloom_filter = column_metadata.__isset.bloom_filter_offset;
+    if (!has_bloom_filter) {
+        // bloom filter is not existed, select all
+        return filtered = false;
+    } else if (get_column_parquet_field() == nullptr) {
+        // which scenerio will reach here? is it unexpected?
+        return filtered = false;
+    }
+
+    bool appliable = check_type_can_apply_bloom_filter(col_type, *get_column_parquet_field());
+    if (appliable) {
+        int32_t offset = column_metadata.bloom_filter_offset;
+        BlockSplitBloomFilter bloom_filter(BlockSplitBloomFilter::mode::PARQUET);
+        //parquet format version>= 2.10 has the bloom filter length info.
+        if (column_metadata.__isset.bloom_filter_length) {
+            RETURN_IF_ERROR(_init_column_bloom_filter(offset, column_metadata.bloom_filter_length, bloom_filter));
+        } else {
+            RETURN_IF_ERROR(_init_column_bloom_filter(offset, 0, bloom_filter));
+        }
+        if (predicates.empty()) {
+            return filtered = false;
+        } else if (pred_relation == CompoundNodeType::AND) {
+            filtered = std::ranges::any_of(predicates, [&bloom_filter](auto& pred) {
+                //TODO: implement parquet_bloom_filter(bloom_filter, col_type, parquet_logical_type, parquet_physical_type) interface,
+                //      and change the value data type to parquet value data type.
+                return pred->support_original_bloom_filter() && !pred->original_bloom_filter(&bloom_filter);
```

Comment:
Check `pred->support_original_bloom_filter()`  first and then init bloom filter is better?